git/git_filter_protocol.go

```
@@ -154,6 +128,40 @@ func (o *ObjectScanner) WriteResponse(outputData []byte) error {
 	return nil
 }
 
+func (o *ObjectScanner) readRequest() (*Request, error) {
+	tracerx.Printf("Process filter command.")
+
+	requestList, err := o.p.readPacketList()
+	if err != nil {
+		return nil, err
+	}
+
+	req := &Request{
+		Header: make(map[string]string),
+	}
+
+	for _, pair := range requestList {
+		v := strings.Split(pair, "=")
+		req.Header[v[0]] = v[1]
+	}
+
+	for {
+		chunk, err := o.p.readPacket()
+		if err != nil {
+			// TODO: should we check the err of this call, to?!
+			o.writeStatus("error")
+			return nil, err
+		}
+		if len(chunk) == 0 {
+			break
+		}
+		req.Payload = append(req.Payload, chunk...) // probably more efficient way?!
+	}
+	o.writeStatus("success")
```

Another ignored error...